scripts/generate_duplicate_sheet.py

```diff
@@ -1,53 +1,82 @@
 # -*- coding: utf-8 -*-
 
+import argparse
 import csv
-import itertools
 import json
+import sys
 from datetime import datetime, timedelta
+from itertools import combinations
+from logging import INFO, basicConfig, getLogger
 
-from bugbug import bugzilla
+from bugbug import bugzilla, similarity
 from bugbug.models.duplicate import DuplicateModel
 
-m = DuplicateModel.load("duplicatemodel")
+basicConfig(level=INFO)
+logger = getLogger(__name__)
 
 REPORTERS_TO_IGNORE = {"intermittent-bug-filer@mozilla.bugs", "wptsync@mozilla.bugs"}
 
 
-try:
-    with open("duplicate_test_bugs.json", "r") as f:
-        test_bugs = json.load(f)
-except FileNotFoundError:
-    test_bug_ids = bugzilla.get_ids_between(
-        datetime.now() - timedelta(days=21), datetime.now()
+def parse_args(args):
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--similaritymodel", default=None)
+    return parser.parse_args(args)
+
+
+def main(args):
+    similarity_model = (
+        similarity.download_and_load_similarity_model(args.similarity)
+        if args.similaritymodel
+        else None
     )
-    test_bugs = bugzilla.get(test_bug_ids)
-    test_bugs = [
-        bug for bug in test_bugs.values() if not bug["creator"] in REPORTERS_TO_IGNORE
-    ]
-    with open("duplicate_test_bugs.json", "w") as f:
-        json.dump(test_bugs, f)
+    duplicate_model = DuplicateModel.load("duplicatemodel")
+    try:
+        with open("duplicate_test_bugs.json", "r") as f:
+            test_bugs = json.load(f)
+    except FileNotFoundError:
+        test_bug_ids = bugzilla.get_ids_between(
+            datetime.now() - timedelta(days=21), datetime.now()
+        )
+        test_bugs = bugzilla.get(test_bug_ids)
+        test_bugs = [
+            bug
+            for bug in test_bugs.values()
+            if not bug["creator"] in REPORTERS_TO_IGNORE
+        ]
+        with open("duplicate_test_bugs.json", "w") as f:
+            json.dump(test_bugs, f)
 
-bug_tuples = list(itertools.combinations(test_bugs, 2))
+    with open("duplicate_predictions.csv", "w") as csvfile:
+        spamwriter = csv.writer(csvfile)
 
-# Warning: Classifying all the test bugs takes a while
-probs = m.classify(bug_tuples, probabilities=True)
+        spamwriter.writerow(
+            ["bug 1 ID", "bug 1 summary", "bug 2 ID", "bug 2 summary", "prediction"]
+        )
+        if similarity_model:
+            test_tuple = []
+            for test_bug in test_bugs:
+                similar_bug_ids = similarity_model.get_similar_bugs(test_bug)
+                sim_bugs = bugzilla.get(similar_bug_ids)
+                test_tuple += [
+                    (sim_bugs[similar_bug_id], test_bug) for similar_bug_id in sim_bugs
```

Comment:
mmmh, shouldn't this be `for similar_bug_id in similar_bug_ids`? Make sure you test the script

Reply:
That won't make a difference, since it'll give the same result. sim_bugs is a dictionary mapping bug id to bug, and similar_bug_ids is a list of ids, so iterating through either of them finally just iterates through bug ids.

Reply:
Right, sorry!
You can rewrite it as this though, which is a bit nicer:
```
                bug_tuples += [
                    (similar_bug, test_bug)
                    for similar_bug in similar_bugs.values()
                ]
```

Reply:
Also, the opposite order (`(test_bug, similar_bug)`) should make the CSV output look nicer